crux_core/src/middleware/effect_handling.rs

```diff
@@ -39,8 +39,8 @@ where
     fn try_process_effect_with(
         &self,
         effect: Effect,
-        resolve_callback: impl FnOnce(
-            RequestHandle<<Self::Op as Operation>::Output>,
+        resolve_callback: impl Fn(
```

Comment:
Can this be `impl FnMut`, which is less restrictive than `Fn`?

Reply:
This is how I think about it. `FnOnce` is the widest set, because it also includes closures that can only be called once because they move data out into their environment. `FnMut` is the next widest because it includes closures that mutate their environment. `Fn` only includes closures that don't move out or mutate. And finally `fn()` is the most restrictive because it's a single type.